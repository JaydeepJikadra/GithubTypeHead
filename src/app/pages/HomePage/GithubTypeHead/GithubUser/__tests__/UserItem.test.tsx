import * as React from 'react';
import { render } from '@testing-library/react';
import { UserItem } from '../UserItem';

const renderUserItem = (props: Parameters<typeof UserItem>[number]) =>
  render(<UserItem {...props} />);

describe('<UserItem />', () => {
  it('should match the snapshot', () => {
    const userItem = renderUserItem({
      login: 'JaydeepJikadra',
      url: 'https://api.github.com/users/JaydeepJikadra',
      avatar_url: 'https://avatars.githubusercontent.com/u/24971844?v=4',
    });
    expect(userItem.container.firstChild).toMatchSnapshot();
  });
});
