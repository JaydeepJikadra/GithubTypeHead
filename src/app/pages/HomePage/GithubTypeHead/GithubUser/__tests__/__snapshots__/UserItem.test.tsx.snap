// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> should match the snapshot 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c4:active {
  opacity: 0.4;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
  min-height: 2.75rem;
  font-weight: 500;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.625rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  width: 8%;
  margin-right: 10px;
}

<div
  class="c0"
>
  <img
    class="c1"
    src="https://avatars.githubusercontent.com/u/24971844?v=4"
  />
  <div
    class="c2"
  >
    JaydeepJikadra
  </div>
  <div
    class="c3"
  >
    <a
      class="c4"
      href="https://api.github.com/users/JaydeepJikadra"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg>
        new-window.svg
      </svg>
    </a>
  </div>
</div>
`;
